### ISSUE TITLE
Provide consistent ordering of key properties (for the schema not the document)

### ISSUE THREAD

**Issue by sthagen** [oasis-tcs/csaf#110]
Now that the schema is quite stable we should consistently order the most common mostly technical "meta" elements like:
```
    title
    description
    type
    pattern
    maxLength
    minLength
    maxItems
    minItems
    maxProperties
    minProperties
    required
    properties
    examples
```

Proposal:
> Let us always provide this order on top of every element:
>```
>    title
>    description
>    type
>```
> Depending on type (object, array, string) let us continue as applicable according to:
>```
>    enum
>    min.*
>    max.*
>    pattern
>    required
>    examples
>    properties
>```

Rationale:
> Simplify reading of the schema by shaping the expectation of the reader and work around the low intensity of the "signal" that a missing minWhatever or maxWhatever quantifier indicates possible absence or inifinte abundance of such an element ...

**Comment by sthagen** [comment]
@tschmidtb51

**Comment by sthagen** [comment]
Updated proposal from hint of @tschmidtb51

**Comment by tschmidtb51** [comment]
> Depending on type (object, array, string) let us continue as applicable according to:
>
> ```   
>   enum
>   min.*
>   max.*
>   pattern
>   required
>   examples
>   properties
>```

@sthagen: To be consistent I suggest using:
```   
   default
   enum
   format   
   pattern
   min.*
   max.*
   required
   properties
   examples
```

**Comment by tschmidtb51** [comment]
For PR #125 I used the following (inclusive) list:
```
    $schema
    $id
    $comment (to be removed before publication)
    title
    description
    dynamicAnchor
    $vocabulary
    allOf
    type
    $ref
    $defs
    default
    enum
    format
    pattern
    const
    maximum
    minimum
    multipleOf
    maxLength
    minLength
    maxItems
    minItems
    maxProperties
    minProperties
    required
    dependentRequired
    uniqueItems
    properties
    items
    if
    then
    else
    not
    unevaluatedItems
    additionalProperties
    unevaluatedProperties
    examples
```

**Comment by sthagen** [comment]
Anything reading less random is good for me 😉

**Comment by sthagen** [comment]
Closed by PR #194
